### PR TITLE
feat(codegen): add --skipfiles flag for DB-only runs (#2440)

### LINKED ISSUE #2440: CodeGen: add inverse of --skipdb (DB-only / --skip-files flag)
## Summary

`npx mj codegen` has a `--skipdb` flag that runs file generation only. There is **no inverse** — no way to run DB writes only while skipping file generation.

## Current state

- **File-only mode** (exists): `npx mj codegen --skipdb`
  - Flag: [packages/MJCLI/src/commands/codegen/index.ts:37-39](https://github.com/MemberJunction/MJ/blob/next/packages/MJCLI/src/commands/codegen/index.ts#L37-L39)
  - Config: `skip_database_generation` in `mj.config.cjs` ([config.ts:456](https://github.com/MemberJunction/MJ/blob/next/packages/CodeGenLib/src/Config/config.ts#L456))
  - Skips SQL execution, permissions, custom scripts; regenerates only TS entities, Angular forms, GraphQL resolvers, actions

- **DB-only mode** (missing): nothing equivalent
  - Only workaround is nulling output directory paths in config — ugly, easy to break

## Proposed

Add `--skip-files` (or `--db-only`) CLI flag that mirrors the `--skipdb` architecture. Also expose as `skip_file_generation` config option.

## Why this matters

1. **Migration-dependent DB touch-ups** — re-apply SPs / views / permissions after a schema migration without thrashing the TypeScript output tree
2. **CI pipelines** — DB smoke tests that don't need the files refreshed
3. **Reset-and-rebuild edge cases** — when the generated TS tree has stub files in place (e.g., bootstrap package chicken-and-egg), running full CodeGen fails during the file-gen phase. DB-only would let you get the DB sync'd first, then regenerate files separately
4. **Parallel workflow separation** — devs working on DB schema changes can regenerate DB state without disturbing in-progress TS source edits

## Ideal end-state

Three independent modes available via flags:
- `npx mj codegen` → both phases (current default)
- `npx mj codegen --skipdb` → files only (current)
- `npx mj codegen --skip-files` → DB only (new)

Optionally, consider splitting the pipeline into separate subcommands:
- `npx mj codegen db` → DB phase only
- `npx mj codegen files` → file phase only
- `npx mj codegen all` → both

## Implementation sketch

In `packages/MJCLI/src/commands/codegen/index.ts`, add `--skipfiles` flag. In `packages/CodeGenLib/src/generator.ts` (or wherever the main pipeline is), wrap the file-generation phase in a conditional mirror of the existing `skipDb` logic.

### PR DESCRIPTION
**Pull request by MS-BC**
## Summary

Adds `--skipfiles` — the inverse of the existing `--skipdb` — so CodeGen's database and file-generation phases can be driven independently.

Closes #2440.

## Behavior matrix

| Invocation | DB writes | File gen |
|---|:-:|:-:|
| `mj codegen` | ✅ | ✅ |
| `mj codegen --skipdb` | ❌ | ✅ |
| `mj codegen --skipfiles` | ✅ | ❌ |
| `mj codegen --skipdb --skipfiles` | ❌ | ❌ |

## Changes

- **`packages/MJCLI/src/commands/codegen/index.ts`** — added `--skipfiles` `Flags.boolean`, new example, and updated description. Also fixed a stale `this.flags.skipDb` (camelCase) reference that didn't match the flag key `skipdb` — the previous code was always passing `undefined`, i.e. the flag never actually did anything from the CLI. Now passes both flags to `runMemberJunctionCodeGeneration`.
- **`packages/CodeGenLib/src/Config/config.ts`** — added `skip_file_generation` (default `false`) to both default settings arrays, so it can also be controlled via `mj.config.cjs`.
- **`packages/CodeGenLib/src/runCodeGen.ts`** — extended signatures of `runMemberJunctionCodeGeneration`, `Run`, `RunInProcess`, and `executeCodeGenPipeline` with a `skipFileGeneration: boolean = false` param. Wrapped the ~130-line file-generation block (GraphQL resolvers, entity subclasses, Angular, DBSchemaJSON, action subclasses) in `if (!skipFiles) { ... } else { warnSpinner('Skipping file generation (skip_file_generation = true)'); SQLLogging.finishSQLLogging(); }`. Signature param is additive (optional with default), so existing callers are unaffected.

## Test plan

- [x] `@memberjunction/codegen-lib` unit tests pass (386/386)
- [x] `@memberjunction/cli` unit tests pass (105/105)
- [x] `npx turbo build --filter=@memberjunction/cli` succeeds (126/126 tasks)
- [x] `npx mj codegen --help` shows `--skipfiles` flag, description, and new example
- [x] Compiled JS confirmed to contain `if (!skipFiles)` gate + paired `SQLLogging.finishSQLLogging()` in else branch
- [ ] Manual run: `npx mj codegen --skipfiles` on a real DB — confirms SQL phase runs, no generated files touched
- [ ] Manual run: `npx mj codegen` (default) still runs both phases unchanged

## Backwards compatibility

- Default behavior is unchanged — both phases run when neither flag is set.
- Existing `--skipdb` users see no behavior change.
- New param on `runMemberJunctionCodeGeneration`, `Run`, `RunInProcess`, and `executeCodeGenPipeline` is optional (defaults `false`), so programmatic callers do not need updates.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/CodeGenLib/src/Config/config.ts [hunk 1]
```diff
@@ -454,6 +454,7 @@ const configInfoSchema = z.object({
   settings: settingInfoSchema.array().default([
     { name: 'mj_core_schema', value: '__mj' },
     { name: 'skip_database_generation', value: false },
+    { name: 'skip_file_generation', value: false },
```
Comment (AN-BC) [inline packages/CodeGenLib/src/Config/config.ts:457]: we should update the default mj.config.cjs we ship to have this new variable also with standard default of false but that will make it easier for people to adjust that way too instead of just CLI.

File: packages/CodeGenLib/src/runCodeGen.ts [hunk 2]
```diff
@@ -281,138 +281,144 @@ export class RunCodeGenBase {
...
-        logStatus('Excluded ' + excludedCount + ' entities from code generation by schema: ' + schemaDetails);
-      }
-
-      const coreEntities = includedEntities.filter(
-        (e) => e.SchemaName.trim().toLowerCase() === mjCoreSchema.trim().toLowerCase()
-      );
-      const nonCoreEntities = includedEntities.filter(
-        (e) => e.SchemaName.trim().toLowerCase() !== mjCoreSchema.trim().toLowerCase()
-      );
+      const skipFiles = skipFileGeneration || getSettingValue('skip_file_generation', false);
```
Comment (AN-BC) [inline packages/CodeGenLib/src/runCodeGen.ts:284]: This is a dirty diff in that the content below `if (!skipFiles) {` is, I think, identical to what it was before, makes it harder to review, which isn't good. More important than that, it could easily contain a regression, so change this to be a minimal diff where you should only see the outer `if` block and the stuff inside should be unmodified or lightly modified I think?

File: packages/CodeGenLib/src/runCodeGen.ts [hunk 3]
```diff
@@ -281,138 +281,144 @@ export class RunCodeGenBase {
...
-      const angularCoreEntitiesOutputDir = outputDir('AngularCoreEntities', false);
-      if (angularCoreEntitiesOutputDir) {
-        if (isVerbose) startSpinner('Generating Angular CORE Entities Code...');
-        const angularGenerator = MJGlobal.Instance.ClassFactory.CreateInstance<AngularClientGeneratorBase>(AngularClientGeneratorBase)!;
-        if (!(await angularGenerator.generateAngularCode(coreEntities, angularCoreEntitiesOutputDir, 'Core', currentUser))) {
-          failSpinner('Error generating Angular CORE Entities code');
-          return false;
-        } else if (isVerbose) succeedSpinner('Angular CORE Entities Code generated');
-      }
+        const angularOutputDir = outputDir('Angular', false);
```
Comment (AN-BC) [inline packages/CodeGenLib/src/runCodeGen.ts]: same dirty diff comment, always focus on minimal diffs when you submit a PR, this is so much harder to (a) review and (b) know with certainty that there isn' a regression. LLM inference to copy stuff like this can be very easily prone to error.

File: packages/CodeGenLib/src/runCodeGen.ts [hunk 4]
```diff
@@ -281,138 +281,144 @@ export class RunCodeGenBase {
...
-      const coreActionsOutputDir = outputDir('CoreActionSubclasses', false);
-      await ActionEngineBase.Instance.Config(false, currentUser);
-      if (coreActionsOutputDir) {
-        if (isVerbose) startSpinner('Generating CORE Actions Code...');
-        const actionsGenerator = MJGlobal.Instance.ClassFactory.CreateInstance<ActionSubClassGeneratorBase>(ActionSubClassGeneratorBase)!;
-        if (!(await actionsGenerator.generateActions(ActionEngineBase.Instance.CoreActions, coreActionsOutputDir))) {
-          failSpinner('Error generating CORE Actions code');
-          return false;
-        } else if (isVerbose) succeedSpinner('CORE Actions Code generated');
+        const actionsOutputDir = outputDir('ActionSubclasses', false);
```
Comment (AN-BC) [inline packages/CodeGenLib/src/runCodeGen.ts]: same comments here and elsewhere on dirty diff

**Review by AN-BC — COMMENTED** [review]
logic seems sound but see comments on dirty diff and make this part of your own review process @MS-BC - otherwise so much harder to spot issues

**Comment by MS-BC** [comment]
Addressed review comments in 18271adb88:

- **Dirty diff fix:** Pulled the 130-line file-generation block out of the `if (!skipFiles) { ... }` wrapper and into a new `runFileGenerationPhase` method. The generator logic is literally unchanged — `git diff` no longer shows any of those 130 lines as modified, so regression review is trivial. Net diff dropped from ~270 lines to ~85.
- **Shipped config:** Added `{ name: 'skip_file_generation', value: false }` to `packages/MJCLI/mj.config.cjs` so users get the toggle in their generated config.

CodeGenLib tests still 386/386 green.

**Review by AN-BC — APPROVED** [review]
